addOns/pscan/src/main/java/org/zaproxy/addon/pscan/internal/StatsPassiveScanner.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2024 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.pscan.internal;
+
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.stream.Collectors;
+import net.htmlparser.jericho.Source;
+import org.apache.commons.httpclient.URIException;
+import org.apache.commons.lang3.StringUtils;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.control.AddOn;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+import org.zaproxy.zap.model.SessionStructure;
+import org.zaproxy.zap.utils.Stats;
+
+public class StatsPassiveScanner extends PluginPassiveScanner {
+
+    public static final String CODE_STATS_PREFIX = "stats.code.";
+    public static final String CONTENT_TYPE_STATS_PREFIX = "stats.contentType.";
+    public static final String RESPONSE_TIME_STATS_PREFIX = "stats.responseTime.";
+
+    private static final boolean LOAD_RULE =
+            org.zaproxy.zap.extension.pscan.scanner.StatsPassiveScanner.class.getAnnotation(
+                            Deprecated.class)
+                    != null;
+    private static StatsPassiveScanner instance;
+
+    // Package to prevent the scan rule from being picked automatically for the manifest, while the
+    // code is being migrated from core.
+    StatsPassiveScanner() {
+        // Keep same status as when in core.
+        setStatus(AddOn.Status.release);
+    }
+
+    @Override
+    public String getName() {
+        return Constant.messages.getString("pscan.stats.passivescanner.title");
```

Comment:
Should we move / copy this property to this add-on's properties file?

Reply:
Done.

Reply:
Thanks, but it has conflicts now, since the other PR was merged.

Reply:
Conflict addressed.